streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -678,12 +678,16 @@ void suspend(final boolean clean,
                 throw taskMigratedException;
             }
         } else {
-            maybeAbortTransactionAndCloseRecordCollector(isZombie);
+            flushState();
```

Comment:
We should add a comment why this is necessary -- otherwise, we might "optimize it away" later on. Or even better, we should add a unit test!

Reply:
Ack